tests/integration/queries/test_workflow_component.py

```diff
@@ -54,3 +54,135 @@ def test_delete_workflow_component(
+    labelset_name = "test-filtered-classes"
+    after_component_id = next(c.id for c in wf.components if c.component_type == "INPUT_OCR_EXTRACTION")
+    source_column_id = org_annotate_dataset.datacolumn_by_name(
+        "News Headlines w/Company Names"
+    ).id
+    modelgroupreq = AddModelGroupComponent(
+        name=mg_name,
+        dataset_id=org_annotate_dataset.id,
+        workflow_id=wf.id,
+        after_component_id=after_component_id,
+        source_column_id=source_column_id,
+        new_labelset_args=NewLabelsetArguments(
+            name=labelset_name,
+            task_type=ModelTaskType.CLASSIFICATION_MULTIPLE,
+            target_names=["type 1", "type 2", "type 3"],
+            datacolumn_id=source_column_id,
+        ),
+    )
+    wf = client.call(modelgroupreq)
+    mg = next(c for c in wf.components if c.component_type == 'MODEL_GROUP')
+    after_component_id = mg.id
+    
+    classes_to_filter = [
+            ["type 1"]
+        ]
+    filtered = AddLinkClassificationComponent(
+            workflow_id=wf.id,
+            after_component_id=after_component_id,
+            model_group_id=mg.model_group.id,
+            filtered_classes=classes_to_filter,
+            labels="actual",
+        )
+    wf = client.call(filtered)
+    new_component =  next(c.id for c in wf.components if c.component_type == 'LINK_CLASSIFICATION_MODEL')
+    assert new_component is not None
+
+
+def test_add_bad_syntax_filtered_classes(indico, org_annotate_dataset):
+    client = IndicoClient()
+    workflowreq = CreateWorkflow(
+        dataset_id=org_annotate_dataset.id, name=f"OrgAnnotate-test-{int(time.time())}"
+    )
+    wf = client.call(workflowreq)
+    mg_name = f"TestAnnotationModelGroup-{int(time.time())}"
+    labelset_name = "test-filtered-classes"
+    after_component_id = next(c.id for c in wf.components if c.component_type == "INPUT_OCR_EXTRACTION")
+    source_column_id = org_annotate_dataset.datacolumn_by_name(
+        "News Headlines w/Company Names"
+    ).id
+    modelgroupreq = AddModelGroupComponent(
+        name=mg_name,
+        dataset_id=org_annotate_dataset.id,
+        workflow_id=wf.id,
+        after_component_id=after_component_id,
+        source_column_id=source_column_id,
+        new_labelset_args=NewLabelsetArguments(
+            name=labelset_name,
+            task_type=ModelTaskType.CLASSIFICATION_MULTIPLE,
+            target_names=["type 1", "type 2", "type 3"],
+            datacolumn_id=source_column_id,
+        ),
+    )
+    wf = client.call(modelgroupreq)
+    mg = next(c for c in wf.components if c.component_type == 'MODEL_GROUP')
+    after_component_id = mg.id
+    
+    classes_to_filter = [
+            ["type 1"],
+            ["type 2"]
+        ]
+    filtered = AddLinkClassificationComponent(
+            workflow_id=wf.id,
+            after_component_id=after_component_id,
+            model_group_id=6108,
+            filtered_classes=[classes_to_filter],
```

Comment:
Ok, so was the problem that Deloitte was passing the filtered classes as a nested list, `[["type 1", "type 1"]]` instead of `["type 1", "type 2"]`?

Reply:
It was actually a nested list inside a nested list [[["type 1", "type2"],["type3]]]